Add Iotty

pburgio opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->



## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [x] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/103073
- Link to parent pull request in the Brands repository:  https://github.com/home-assistant/brands/pull/4966
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced comprehensive documentation for integrating iotty smart devices with Home Assistant, enhancing user experience and installation ease.
  - Detailed instructions for connecting devices such as Smart Switch, Smart Shades Switch, and Smart Outlet, including links to US and EU versions.

- **Documentation**
  - Added a new markdown file that serves as a crucial resource for users looking to incorporate iotty devices into their Home Assistant setup.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

pburgio committed:
Added iotty documentation

pburgio committed:
Merge branch 'current' into iotty-initial-docs

pburgio committed:
Merge branch 'next' into iotty-initial-docs

pburgio committed:
Merge branch 'next' into iotty-initial-docs

pburgio committed:
Merge branch 'next' into iotty-initial-docs

pburgio committed:
Merge branch 'next' into iotty-initial-docs

pburgio committed:
Merge branch 'next' into iotty-initial-docs

pburgio committed:
Merge branch 'next' into iotty-initial-docs

pburgio committed:
Merge branch 'next' into iotty-initial-docs

pburgio committed:
Merge branch 'next' into iotty-initial-docs

pburgio committed:
Merge branch 'next' into iotty-initial-docs

pburgio committed:
Merge branch 'next' into iotty-initial-docs

pburgio committed:
Merge branch 'next' into iotty-initial-docs

pburgio committed:
Merge branch 'next' into iotty-initial-docs

c0ffeeca7 commented on source/_integrations/iotty.markdown:
```diff
@@ -0,0 +1,42 @@
+---
+title: iotty
+description: "Instructions on how to connect iotty Smart Devices to Home Assistant."
+ha_release: "0.38"
```
```suggestion
ha_release: "2024.8"
```
This is the release when the integration was first introduced. For more info on the header, refer to the documentation on [creating a new integration documentation page](https://developers.home-assistant.io/docs/documenting/create-page)

c0ffeeca7 commented on source/_integrations/iotty.markdown:
```diff
@@ -0,0 +1,42 @@
+---
+
+The iotty family is focused on beautiful aesthetics and includes the Smart Switch for lights and gates, the Smart Shades Switch for blinds and shutters and the Smart Outlet. Install the iotty devices and enjoy your new Smart Home with ease!
+
```
```suggestion

## Supported devices

```

c0ffeeca7 committed:
Update release version and add header to supported devices section

c0ffeeca7 commented on source/_integrations/iotty.markdown:
```diff
@@ -0,0 +1,44 @@
+  EU VERSION:
+  - https://iotty.fr/collections/frontpage/products/prise-intelligent-iotty
+  - https://iotty.de/collections/frontpage/products/iotty-smart-steckdose
+  - https://iotty.it/collections/frontpage/products/iotty-plus-presa-intelligente
```
```suggestion
  - https://iotty.it/collections/frontpage/products/iotty-plus-presa-intelligente
  
  {% include integrations/config_flow.md %}
```

c0ffeeca7 committed:
Add config flow

c0ffeeca7 commented on source/_integrations/iotty.markdown:
```diff
@@ -0,0 +1,46 @@
+ha_config_flow: true
+ha_codeowners:
+  - '@pburgio'
+ha_domain: iotty
```
```suggestion
ha_domain: iotty
ha_platforms:
  - switch
 ha_integration_type: hub
```

c0ffeeca7 committed:
add platform and integrration type to header

c0ffeeca7 commented on source/_integrations/iotty.markdown:
```diff
@@ -0,0 +1,49 @@
+ha_domain: iotty
+ha_platforms:
+  - switch
+ ha_integration_type: hub
```
```suggestion
ha_integration_type: hub
```

c0ffeeca7 committed:
Update source/_integrations/iotty.markdown

c0ffeeca7 commented on source/_integrations/iotty.markdown:
```diff
@@ -0,0 +1,49 @@
+  - https://iotty.fr/collections/frontpage/products/prise-intelligent-iotty
+  - https://iotty.de/collections/frontpage/products/iotty-smart-steckdose
+  - https://iotty.it/collections/frontpage/products/iotty-plus-presa-intelligente
+  
```
Are there any prerequisites? Anything I need to do or have before I can install the iotty integration, for example, do I need an iotty account and login credentials?

shapournemati-iotty replied:
Done

c0ffeeca7 commented on source/_integrations/iotty.markdown:
```diff
@@ -0,0 +1,49 @@
+title: iotty
+description: "Instructions on how to connect iotty Smart Devices to Home Assistant."
+ha_release: "2024.8"
+ha_category: "No score"
```
```suggestion
ha_category: 
  - Switch
```

c0ffeeca7 committed:
Add switch to category

https://github.com/home-assistant/core/pull/103073/files#diff-42fa91b71097e4c7eebdd723fe0c616a86d1356ac85186fc4ed4adad918bd2d9R1

c0ffeeca7 commented on source/_integrations/iotty.markdown:
```diff
@@ -0,0 +1,50 @@
+ha_integration_type: hub
+---
+
+The iotty family is focused on beautiful aesthetics and includes the Smart Switch for lights and gates, the Smart Shades Switch for blinds and shutters and the Smart Outlet. Install the iotty devices and enjoy your new Smart Home with ease!
```
```suggestion
The iotty {% term integration%} lets you integrate iotty devices into Home Assistant. The iotty family includes the smart switch for lights and gates, the smart shades switch for blinds and shutters, and the smart outlet. 
```
Small tweak to focus the intro more on what the integration does.

c0ffeeca7 commented on source/_integrations/iotty.markdown:
```diff
@@ -0,0 +1,50 @@
+  - https://iotty.de/collections/frontpage/products/iotty-smart-steckdose
+  - https://iotty.it/collections/frontpage/products/iotty-plus-presa-intelligente
+  
+  {% include integrations/config_flow.md %}
```
```suggestion
```

c0ffeeca7 commented on source/_integrations/iotty.markdown:
```diff
@@ -0,0 +1,50 @@
+  - https://iotty.fr/collections/frontpage/products/prise-intelligent-iotty
+  - https://iotty.de/collections/frontpage/products/iotty-smart-steckdose
+  - https://iotty.it/collections/frontpage/products/iotty-plus-presa-intelligente
+  
```
```suggestion
```

c0ffeeca7 commented on source/_integrations/iotty.markdown:
```diff
@@ -0,0 +1,50 @@
+
+The iotty family is focused on beautiful aesthetics and includes the Smart Switch for lights and gates, the Smart Shades Switch for blinds and shutters and the Smart Outlet. Install the iotty devices and enjoy your new Smart Home with ease!
+
+## Supported devices
```
```suggestion

  {% include integrations/config_flow.md %}

## Supported devices
```

c0ffeeca7 committed:
move config_flow up

c0ffeeca7 commented on source/_integrations/iotty.markdown:
```diff
@@ -0,0 +1,51 @@
+  EU VERSION:
+  - https://iotty.fr/collections/frontpage/products/prise-intelligent-iotty
+  - https://iotty.de/collections/frontpage/products/iotty-smart-steckdose
+  - https://iotty.it/collections/frontpage/products/iotty-plus-presa-intelligente
```
Please add a section that gives more information about the supported entities. For example, Husquvarna or La Marzocco have sections on switches, for reference.

shapournemati-iotty replied:
In our integration things are pretty simple: one switch per device. I have added a small section about that, let me know if that is what you were thinking about

c0ffeeca7 committed:
Update source/_integrations/iotty.markdown

c0ffeeca7 commented on source/_integrations/iotty.markdown:
```diff
@@ -0,0 +1,51 @@
+The iotty {% term integration%} lets you integrate iotty devices into Home Assistant. The iotty family includes the smart switch for lights and gates, the smart shades switch for blinds and shutters, and the smart outlet. 
+
+
+  {% include integrations/config_flow.md %}
```
```suggestion
{% include integrations/config_flow.md %}
```

c0ffeeca7 committed:
Update source/_integrations/iotty.markdown

c0ffeeca7 commented on source/_integrations/iotty.markdown:
```diff
@@ -0,0 +1,51 @@
+  - https://iotty.it/collections/frontpage/products/iotty-plus-interruttore-intelligente-per-luci-e-cancelli
+
+
+More integrations are coming soon:
```
```suggestion
```

shapournemati-iotty replied:
Done

c0ffeeca7 commented on source/_integrations/iotty.markdown:
```diff
@@ -0,0 +1,51 @@
+  EU VERSION:
+  - https://iotty.fr/collections/frontpage/products/prise-intelligent-iotty
+  - https://iotty.de/collections/frontpage/products/iotty-smart-steckdose
+  - https://iotty.it/collections/frontpage/products/iotty-plus-presa-intelligente
```
```suggestion
```
I think the documentation should focus on the current status.

shapournemati-iotty replied:
I agree, it has been removed

c0ffeeca7 commented on source/_integrations/iotty.markdown:
```diff
@@ -0,0 +1,51 @@
+---
+
+The iotty {% term integration%} lets you integrate iotty devices into Home Assistant. The iotty family includes the smart switch for lights and gates, the smart shades switch for blinds and shutters, and the smart outlet. 
+
```
Are there any Prerequisites? do I need an account and credentials before I can configure the integration? If so, describe how I get them.

shapournemati-iotty committed:
Add prerequisites, fix links so they are in markdown style, and add supported entities descriptions

shapournemati-iotty commented on source/_integrations/iotty.markdown:
```diff
@@ -0,0 +1,42 @@
+  EU VERSION:
+  - https://iotty.uk/collections/frontpage/products/iotty-smart-switch
+  - https://iotty.uk/collections/frontpage/products/iotty-smart-switch?variant=40820222460082
+  - https://iotty.it/collections/frontpage/products/iotty-plus-interruttore-intelligente-per-luci-e-cancelli
```
Done

shapournemati-iotty commented on source/_integrations/iotty.markdown:
```diff
@@ -0,0 +1,42 @@
+  EU VERSION:
+  - https://iotty.fr/collections/frontpage/products/prise-intelligent-iotty
+  - https://iotty.de/collections/frontpage/products/iotty-smart-steckdose
+  - https://iotty.it/collections/frontpage/products/iotty-plus-presa-intelligente
```
Discarded, as this is no longer present in the documentation

shapournemati-iotty commented on source/_integrations/iotty.markdown:
```diff
@@ -0,0 +1,44 @@
+  - https://iotty.it/collections/frontpage/products/iotty-plus-interruttore-intelligente-per-luci-e-cancelli
+
+
+More integrations are coming soon:
```
Discarded, as this is no longer present in the documentation

shapournemati-iotty committed:
fix spelling of Home Assistant

c0ffeeca7 commented on source/_integrations/iotty.markdown:
```diff
@@ -0,0 +1,49 @@
+  - [iotty Smart Switch (variant)](https://iotty.uk/collections/frontpage/products/iotty-smart-switch?variant=40820222460082)
+  - [iotty Plus Interruttore Intelligente](https://iotty.it/collections/frontpage/products/iotty-plus-interruttore-intelligente-per-luci-e-cancelli)
+
+## Supported Entities
```
```suggestion
## Supported entities
```
We use [sentence-style capitalization](https://learn.microsoft.com/en-us/style-guide/text-formatting/using-type/use-sentence-style-capitalization) also in titles

c0ffeeca7 commented on source/_integrations/iotty.markdown:
```diff
@@ -0,0 +1,49 @@
+
+### Switches
+
+The main, unnamed switch entity, controls the lightswitch, turning it on or off the gang related to it.
```
```suggestion
The main, unnamed, switch entity controls the light switch, turning it on or off the gang related to it.
```

c0ffeeca7 commented on source/_integrations/iotty.markdown:
```diff
@@ -0,0 +1,49 @@
+
+## Prerequisites
+
+In order to use this integration you must have a iotty account, and enter its credentials during account pairing.
```
```suggestion
In order to use this integration, you must have an iotty account, and enter its credentials during account pairing.
```

c0ffeeca7 commented on source/_integrations/iotty.markdown:
```diff
@@ -0,0 +1,49 @@
+## Prerequisites
+
+In order to use this integration you must have a iotty account, and enter its credentials during account pairing.
+To create a iotty account, you need to get the App from the [App Store](https://apps.apple.com/it/app/iotty-smart-home/id1230937401) or [Play Store](https://play.google.com/store/apps/details?id=com.dynamicait.iotty&hl=en).
```
```suggestion
To create an iotty account, you need to get the App from the [App Store](https://apps.apple.com/it/app/iotty-smart-home/id1230937401) or [Play Store](https://play.google.com/store/apps/details?id=com.dynamicait.iotty&hl=en).
```

c0ffeeca7 commented on source/_integrations/iotty.markdown:
```diff
@@ -0,0 +1,49 @@
+
+## Supported devices
+
+This integration currently supports the following iotty Devices:
```
```suggestion
This integration currently supports the following iotty devices:
```

c0ffeeca7 reviewed (changes requested):
Tiny tweaks

c0ffeeca7 committed:
Apply suggestions from code review

c0ffeeca7 commented on source/_integrations/iotty.markdown:
```diff
@@ -0,0 +1,49 @@
+
+## Supported entities
+
+Each iotty device gets mapped into one Home Assistant device per gang, each with its own Switch entity.
```
what do you mean by "per gang"? group?

shapournemati-iotty replied:
Hi, "gang" is a pretty standard term used when referring to te single switches on a plate, as you can see in this wikipedia article: [Light Switch](https://en.wikipedia.org/wiki/Light_switch).

I didn't want to reuse the word "Device" or "Switch" as it would be using the same word to define itself. Let me know if that is clear, or if you have any other suggestion to replace the word "gang"

c0ffeeca7 replied:
Thank you for the explanation. Learned something today.

pburgio committed:
Merge branch 'next' into iotty-initial-docs

c0ffeeca7 reviewed (approved):
Thank you, @pburgio and @shapournemati-iotty  :+1:

MartinHjelmare commented on source/_integrations/iotty.markdown:
```diff
@@ -0,0 +1,49 @@
+
+### Switches
+
+The main, unnamed, switch entity controls the light switch, turning it on or off the gang related to it.
```
This sentence has incorrect grammar. I can't make a suggestion since it's not clear to me what meaning is meant.

shapournemati-iotty replied:
Hej Martin, thanks for the feedback. Since the PR has already been merged, what is the suggested course of action here? Open another PR to fix this? Wait for the next release in which some functionality will be added to my integration?

For what concerns the meaning, what I want to convey is:

1. There is only one entity per device
2. The main (and only) entity of the device is unnamed
3. The switch entity is meant to control a light switch

MartinHjelmare replied:
Please open a new PR and adjust the sentence.
